### ISSUE TITLE
Simulate Hibernation / Suspend-To-Disk

### ISSUE THREAD

**Issue by xloem** [QubesOS/qubes-issues#2414]
I think this is a known idea but I didn't find an issue for it.

For laptops, hibernation is great to just close the lid and come back to your work a week later.  For security, this is important because so much data is in RAM.

Xen does not support suspend-to-disk, but Qubes could emulate suspend-to-disk by hibernating all the VMs, and then simply shutting down.  It would be great if this were formally supported some day.

**Comment by andrewdavidwong** [comment]
Yes, this has come up on the mailing lists quite a lot over the years:

https://groups.google.com/forum/#!searchin/qubes-users/hibernate

My sense is that this is something that is possible but that would be a very low priority (if a priority at all) for the Qubes team. So, we'd be open to accepting patches for this, but that's probably the only way it'll happen (at least any time soon).

**Comment by d2r** [comment]
This would be desirable for the purposes of clearing from RAM the key used for full disk encryption.

**Comment by iamahuman** [comment]
#2273 (xl save/restore) or virsh managedsave might be useful here.

**Comment by ben-grande** [comment]
@andrewdavidwong I believe this is a duplicate of:

- https://github.com/QubesOS/qubes-issues/issues/2273
